src/base/i18n.cc

```diff
@@ -374,9 +374,10 @@ void set_locale(const std::string& name) {
 		// maybe, do another check with the return value (?)
 		locale = lang;
 	}
-
+#if defined(HAS_NL_MSG_CAT_CNTR)
```

Comment:
This is a glibc internal so it'd be better to just #ifdef __GLIBC__, I think. See what I did at https://github.com/anope/anope/pull/258.

Reply:
This would again use an internal define (double `__`), which caused issues with `__WORDSIZE`. So I'd rather go the safe way and try to use a libc independent approach without assumptions about internals. Maybe there is a way to remove this `++_nl_msg_cat_cntr;` completely?

Reply:
You would be checking an internal __ purely to see if you're using an internal, so it's probably fine (because if it was undefined, who cares?) but I take your point.

i.e. _nl_msg_cat_cntr is purely only ever available with glibc combined with GNU gettext.

Interestingly, https://www.gnu.org/software/gettext/manual/html_node/gettext-grok.html seems to define-it-then-use it. I think what you did is okay, I just don't ever expect it to pass on non-glibc, and if it did, it might have unexpected results.

I'll ask the GNU gettext maintainer what he thinks.

Reply:
I agree with Sam: When you use `_nl_msg_cat_cntr` (for the purpose as here, for notifying the gettext cache that the locale is changing), you should test for it through a configure-time test. Otherwise you will risk compilation errors on platforms that don't have it. The only platforms that have it are glibc and Solaris 11 platforms.

Testing `defined(__USE_GNU_GETTEXT)` is not the right way to test for this feature, because `__USE_GNU_GETTEXT` is defined
- on glibc systems
- on musl libc (which does not support several GNU gettext features (charset conversion, ISO C 99 <inttypes.h> format string directives, and the 'I' format string flag)
- when GNU libintl is installed.
Therefore with this test you would get a link error on musl libc and missed functionality on Solaris 11.

Reply:
Thanks for the explanation Bruno, let's go for the configure check then. I'll also see about fixing Anope.

Reply:
Thank you both for the feedback and insights.
So if I understand correctly, you say that I did it already correctly here? cmake checks the presence of `_nl_msg_cat_cntr` during configuration. It does not use any defines, but simply checks, whether it can compile such a source and defines `HAS_NL_MSG_CAT_CNTR` if it can. 

https://github.com/widelands/widelands/blob/c0cda98f81606115f2f722d75c11bc3e9e2d6cf3/src/CMakeLists.txt#L1-L3

Reply:
The `HAS_NL_MSG_CAT_CNTR` approach is the correct one.

But does the check_cxx_source_compiles test actually work in the third case (when GNU libintl is installed)? I don't know what check_cxx_source_compiles does (a compilation check or a compilation+link check?) and whether it considers the need to use `-Lsome/path -libintl` in that case.

Reply:
It does a compile+link check. However libintl was not considered, so it already failed in the Windows builds (see [here](https://github.com/widelands/widelands/actions/runs/4007249397/jobs/6883529931#step:7:111)), so this was a good hint. 
I added it as library for the check, if it is found during the library discovery.